### DISCUSSION TITLE
Qexo添加的tags:[]会导致搜索引擎索引出现问题

### DISCUSSION THREAD

**Discussion by AkinaHaruka** [Qexo/Qexo#178] [Q&A] [answered]
我的文章的标签一直都是这样写的：
![image](https://user-images.githubusercontent.com/53249654/179537012-cb716567-18ac-4fc8-8d0c-fe51347b47df.png)
Hexo可以识别
![image](https://user-images.githubusercontent.com/53249654/179537094-dff2230e-5465-4589-88da-2d3cb8c00bf8.png)

但是新版的Dev分支的Qexo会采用更新的方法：
![image](https://user-images.githubusercontent.com/53249654/179537230-f44af2be-1038-4a51-aa83-5c24f7f00bd6.png)
但是我仍然使用原始方法

目前发现tags: []这个会导致网页中出现<meta name="robots" content="noindex">，导致搜索引擎不索引
证明如下

# 有tags:[]
![33D029B83BB309DCBA3758058D66773C](https://user-images.githubusercontent.com/53249654/179538192-44a22782-aa23-479e-9425-409c5a633866.jpg)

![071487AB8E1C5173A78858E1EBEF09DF](https://user-images.githubusercontent.com/53249654/179538211-b4e4dba8-3311-42ef-a99f-b2e080dcde0b.jpg)

# 无tags:[]
![AFE21195A8560D523BA9CE110E19DE4E](https://user-images.githubusercontent.com/53249654/179538282-7e40d0d2-f883-404e-9a19-a8915aab4820.jpg)

![01B153FD08D42C0B5894AE78067AD6BD](https://user-images.githubusercontent.com/53249654/179538304-c9ffc6cf-47e3-47f8-8b5f-4463d3b048f9.jpg)

**Comment by am-abudu** [comment, marked as answer]
在添加标签后似乎一切正常，但是值得注意

---

根据Hexo官方文档中Front-matter部分，正确的tag写法应为`tags`而非`tag`，且该问题还受主题和插件影响，故不作为Bug处理

**Comment by AkinaHaruka** [comment]
问题发现：
icarus主题：hexo-theme-icarus/layout/common/head.jsx中第五十二行：
![image](https://user-images.githubusercontent.com/53249654/179542869-75dd62fb-afad-40f5-9253-6e6cc95293fd.png)
当getPageTitle方法调用时，会判断标签和分类的状态，当标签或分类为空时，会加上noindex标记

**Comment by AkinaHaruka** [comment]
该问题建议保留，以让遇到相同问题的用户快速找到解决方案
